lxd/backup.go

```diff
@@ -554,3 +564,33 @@ func volumeBackupWriteIndex(s *state.State, projectName string, volumeName strin
 
 	return nil
 }
+
+func pruneExpiredStorageVolumeBackups(ctx context.Context, s *state.State) error {
+	// Get the list of expired backups.
+	backups, err := s.DB.Cluster.GetExpiredStorageVolumeBackups()
```

Comment:
As this is a very specific function, can we modify it to return the volume info as well (or perhaps only) so we avoid needing to load the volume individually as multiple queries.

Also it would be good to load everything we need from the DB in a single transaction (or even better a single query if we just need to pull volumes for the expire backups), and then group them by pool in a map so we dont have to load the pool multiple times potentially.

Reply:
> As this is a very specific function, can we modify it to return the volume info as well (or perhaps only) so we avoid needing to load the volume individually as multiple queries.

I don't see how. In order to delete a backup volume, we need the pool name (from storage pool), volume name and volume project (both from the storage volume), and some fields from the actual volume backup. Only returning the volume info won't provide enough information to delete a backup.

> Also it would be good to load everything we need from the DB in a single transaction (or even better a single query if we just need to pull volumes for the expire backups), and then group them by pool in a map so we dont have to load the pool multiple times potentially.

I understand that a single transaction or query would be easier on the database. But this would either result in a custom struct just for expired storage volume backups or a very unclean map as it would need to include all information mentioned above.

The other solution would be to move some functionality from `*Cluster` to `*ClusterTx`, but I don't believe this belongs in this PR.